src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -510,13 +567,17 @@ private OAuthProxy(
             this.grant = Objects.requireNonNull(grant);
             this.isCorsPreflight = isCorsPreflight;
 
-            if (expiresAtMillis != EXPIRES_NEVER)
+            long delay = 0;
+            if (challengeResponseTimeout > 0)
             {
-                final long delay = expiresAtMillis - System.currentTimeMillis();
-
-                this.expiryFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId, TOKEN_EXPIRED_SIGNAL);
-                this.grant.acquire();
+                delay = expiresAtMillis - challengeResponseTimeout - System.currentTimeMillis();
+            }
+            else if (expiresAtMillis != EXPIRES_NEVER && challengeResponseTimeout == 0)
+            {
+                delay = expiresAtMillis - System.currentTimeMillis();
             }
+            this.signalFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId,
+                    GRANT_VALIDATION_SIGNAL);
```

Comment:
This logic has changed, it used to not schedule a signal if `EXPIRES_NEVER`.
Suggest this instead.
```
assert challengeResponseTimeout >= 0;
if (expiresAtMillis != EXPIRES_NEVER)
{
    final long challengeDelay = expiresAtMillis - challengeResponseTimeout - System.currentTimeMillis();
    this.signalFuture = executor.schedule(delay, MILLISECONDS, targetRouteId, targetStreamId, GRANT_VALIDATION_SIGNAL);
}
```